File: acmeshop/acmeshop/src/main/java/org/acme/shop/model/Customer.java
package org.acme.shop.model;

import io.quarkus.hibernate.orm.panache.PanacheEntity;

import jakarta.persistence.Entity;
import jakarta.persistence.OneToMany;
import jakarta.validation.constraints.Size;
import java.util.ArrayList;
import java.util.List;

@Entity
public class Customer extends PanacheEntity {

    @Size(min = 4)
    private String name;

    @OneToMany
    private List<ShopOrder> orders = new ArrayList<>();

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<ShopOrder> getOrders() {
        return orders;
    }

    public void setOrders(List<ShopOrder> orders) {
        this.orders = orders;
    }

    public void addOrder(ShopOrder order) {
        this.orders.add(order);
        order.setUser(this);
    }
}

File: acmeshop/acmeshop-kotlin/src/main/kotlin/shop/model/Customer.kt
package shop.model

import io.quarkus.hibernate.orm.panache.kotlin.PanacheCompanion
import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntity
import jakarta.persistence.Entity
import jakarta.persistence.OneToMany
import jakarta.validation.constraints.Size

@Entity
class Customer : PanacheEntity() {

    companion object: PanacheCompanion<Customer> {
    }

    @Size(min = 4)
    lateinit var name: String

    @OneToMany
    private var orders: MutableList<ShopOrder> = ArrayList()
    fun getOrders(): List<ShopOrder> {
        return orders
    }

    fun setOrders(orders: MutableList<ShopOrder>) {
        this.orders = orders
    }

    fun addOrder(order: ShopOrder) {
        orders.add(order)
        order.user = this
    }
}